File: src/PostsByMarko.Host/Data/Entities/Post.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PostsByMarko.Host.Data.Entities
{
    public class Post
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public Guid AuthorId { get; set; }
        public User Author { get; set; } = default!;
        public string Title { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime LastUpdatedAt { get; set; } = DateTime.UtcNow;
        public bool Hidden { get; set; } = false;

        public Post() { }
    }
}

File: src/PostsByMarko.Client/src/types/post.ts
import { User } from "./user";

export interface Post {
  id?: string | null;
  authorId?: string | null;
  author?: User | null;
  title: string;
  content: string;
  createdAt?: Date | null;
  lastUpdatedAt?: Date | null;
  hidden: boolean;
}

export interface CreatePostRequest {
  title: string;
  content: string;
}

export interface UpdatePostRequest {
  title: string;
  content: string;
  hidden: boolean;
}
